src/pieces.py

```diff
@@ -107,7 +107,7 @@ def __repr__(self) -> str:
         return " "  # An unspecified piece is a ghost piece.
 
     def __hash__(self):
-        return hash(self.__class__.__name__)
+        return hash("Piece")
 
```

Comment:
This is easier with the `@dataclass(frozen=True)` decorator as it makes them safely-hashable without extra code.